litestar/_signature/models/utils.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal, cast
+
+from litestar.params import ParameterKwarg
+
+if TYPE_CHECKING:
+    from litestar.connection.base import ASGIConnection
+    from litestar.utils.signature import ParsedParameter
+
+
+def get_param_source(key: str | int, connection: ASGIConnection) -> Literal["cookie", "header", "query"] | None:
```

Comment:
So, the contents of this file are not really required as they are now. 

The signature model is responsible for modelling the required kwargs. We than have a pretty similar logic under the `_kwargs` namespace for modelling what you're doing here, basically. Take a look there.

Reply:
Awesome, I'll take a look at that, thanks.